### DISCUSSION TITLE
Is it possible to generate client side logs of traffic to and from GitHub copilot

### DISCUSSION THREAD

**Discussion by mathewbutler** [Genuitec/Copilot4Eclipse-community#32] [Help] [answered]
My current project is asking for some assurance of the data being sent to, and received from GitHub. Specifically the project is is asking for the generation of client side logs. 

Can you please advise if this is possible using Copilot4Eclipse , how this is achieved, and any associated caveats (I assume a performance hit)? 

Operation System: windows 10 and 11
Eclipse version: 2024-06 (4.32.0)
Oracle Java 17 update 11.

Copilot4Eclipse 1.2.1.202406240349	

Thanks in advance.

**Comment by wayneparrott** [comment, marked as answer]
Yes, this is possible. But as you identified the logging can be extensive. 
Check out this doc: http://localhost:3000/products/copilot4eclipse/docs/basics/logging#logging-communications-with-github-copilot-services-low-level

Let use know if this works for you?

Wayne
Copilot4Eclipse Team

**Reply by mathewbutler** [reply]
Appreciate the response - thank you. I will update when have had a closer look.

Leaving an externally visible link here in case this helps anyone - [https://www.genuitec.com/products/copilot4eclipse/docs/basics/logging/#logging-communications-with-github-copilot-services-low-level](https://www.genuitec.com/products/copilot4eclipse/docs/basics/logging/#logging-communications-with-github-copilot-services-low-level)

**Reply by mathewbutler** [reply]
I'm being asked if these logs could be protected by use of an append only files system to prevent delete or modification other than adding additional writes. I've suggested that we test this. 

Is this a file system target configured for append only support for use by Github Copilot and Copilot4Eclipse?

**Reply by wayneparrott** [reply]
The Copilot language server logging appends output to a single log file. The log file is created under the current Eclipse workspace folder with the user's id as owner and its location is not configurable at this time.  If you have special requirements you can post here or consult with us at support@genuitec.com. 

Wayne
Copilot4Eclipse Team
